test_parser.py

```diff
@@ -54,7 +55,7 @@ def test_parser(zone, data_type, target_datetime):
 
     elapsed_time = time.time() - start
     if isinstance(res, (list, tuple)):
-        res_list = iter(res)
+        res_list = list(res)
```

Comment:
I think you can keep it an iterable?

Reply:
Can't iterables only be used once? And we have two checks on this data now, so if that is the case, then no.

Reply:
good point - agree